datahub-web-react/src/app/homeV3/modules/assetCollection/AssetCollectionModule.tsx

```diff
@@ -20,19 +23,72 @@ const AssetCollectionModule = (props: ModuleProps) => {
         [props.module.properties.params.assetCollectionParams?.assetUrns],
     );
 
+    const dynamicFilterLogicalPredicate: LogicalPredicate | undefined = useMemo(
+        () =>
+            props.module.properties.params.assetCollectionParams?.dynamicFilterJson
+                ? JSON.parse(props.module.properties.params.assetCollectionParams?.dynamicFilterJson)
+                : undefined,
+        [props.module.properties.params.assetCollectionParams?.dynamicFilterJson],
+    );
+
+    const shouldFetchByDynamicFilter = useMemo(
+        () => assetUrns.length === 0 && !!dynamicFilterLogicalPredicate,
+        [assetUrns, dynamicFilterLogicalPredicate],
+    );
```

Comment:
it makes me nervous if we get into a situation where somehow assetUrns is not 0 and we have a dynamic filter. maybe we could add a check in the graphql api or something to make sure we're never upserting a module with both of these filled out?